backend/src/nodes/ncnn_nodes.py

```diff
@@ -25,6 +25,24 @@
 from .model_save_nodes import NcnnSaveNode
 
 
+class AllocatorContextManager:
```

Comment:
This still isn't correct. If the second allocator fails for some reason, then the first one doesn't get cleared. If either one fails, then its field will be None and `__exit__` will deref null.

I would do it like this:

```py
from contextlib import contextmanager

@contextmanager
def managed_blob_vkallocator(vkdev: ncnn.VulkanDevice):
    try:
        blob_vkallocator = vkdev.acquire_blob_allocator()
    except:
        blob_vkallocator = ncnn.VkBlobAllocator(vkdev)
    try:
        yield blob_vkallocator
    finally:
        blob_vkallocator.clear()

@contextmanager
def ncnn_allocators(vkdev: ncnn.VulkanDevice):
    with managed_blob_vkallocator(vkdev) as blob_vkallocator:
        try:
            staging_vkallocator = vkdev.acquire_staging_allocator()
        except:
            staging_vkallocator = ncnn.VkStagingAllocator(vkdev)
        try:
            yield blob_vkallocator, staging_vkallocator
        finally:
            staging_vkallocator.clear()
```

More verbose, yes, but it's guaranteed to call `clear`.

Also, what does `clear` do anyway? I assumed that it's leaking memory if it doesn't get called.

Reply:
Correct. It holds onto vram without it